app/views/products/_form.html.haml

```diff
@@ -0,0 +1,40 @@
+= form_for @product do |f|
+  .col-md-6.form-group
+    = f.label :source_id, t('activerecord.attributes.product.source_id')
+    = f.text_field :source_id, class: "form-control"
+  .col-md-6.form-group
+    = f.label :title, t('activerecord.attributes.product.title')
+    = f.text_field :title, class: "form-control"
+  .col-md-12.form-group
+    = f.label :is_active, t('activerecord.attributes.product.is_active')
+    = f.check_box :is_active
+  .col-md-6.form-group
+    = f.label :aff_code, t('activerecord.attributes.product.aff_code')
+    = f.text_field :aff_code, class: "form-control"
+  .col-md-6.form-group
+    = f.label :price, t('activerecord.attributes.product.price')
+    = f.text_field :price, class: "form-control"
+  .col-md-6.form-group
+    = f.label :old_price, t('activerecord.attributes.product.old_price')
+    = f.text_field :old_price, class: "form-control"
+  .col-md-6.form-group
+    = f.label :short_message, t('activerecord.attributes.product.short_message')
+    = f.text_field :short_message, class: "form-control"
+  .col-md-12
+    = f.label :description, t('activerecord.attributes.product.description')
+    = f.text_area :description, class: "form-control"
+
+
+  .col-md-12
+    %hr/
+  .col-md-4.form-group
+    = f.label :campaign_id, t('activerecord.attributes.product.campaign')
+    = f.select :campaign_id, options_for_select(Campaign.all.map{|c| [c.name, c.id]}), {}, class: "form-control", placeholder: t("activerecord.attributes.product.campaign")
+  .col-md-4.form-group
+    = f.label :widget_id, t('activerecord.attributes.product.widget')
+    = f.select :widget_id, options_for_select(Widget.all.map{|c| [c.name, c.id]}), {}, class: "form-control", placeholder: t("activerecord.attributes.product.widget")
```

Comment:
For Campaign we can use a modal on same page to create a campaign.

And for Widget and Brand(because they have only one field) we can use selectize.js. This library has the option to create a new record if you want, very fast.